learn/reif_examples.dj

```diff
@@ -0,0 +1,453 @@
+
+We will use tmember_t to demonstrate an example.   T is bound to true when a condition holds, and false  when it doesn't. 
+
+``` prolog
+?- tmember_t(#<(1),[1,2,3,4,6],T).
+   T = true.
+?- tmember_t(#<(100),[1,2,3,4,6],T).
+   T = false.
+?- 
+```
+
+The prefix of t_  (tmember/2,tmember_t/3, tfilter, tpartition/4)  in a predicate name indicates the first 
+argument is  a predicate that takes as as an argument a predicate  (or partially applied predicate) taking additonal arguments, the last of which will
+be a reified boolean value for whether the query holds.  
+
+The naming convention for the parameters which are predicates is C_N, where C may convey some meaning about the arguments purpose, and the 
+N indicates how many additional arguments are required by the  predicate.
+
+A suffix of _t indicates the last parameter of a predicate is a reified boolean value.  The last argument of the following are also reifed boolean values:  
+(=)/3, (,)/3, (;)/3, dif/3.  
+
+
+You may wish to follow that convention when writing your own predicates.  
+
+It is useful to inspect the source of reif so you can see the information conveyed in the predicate and predicate parameter names.
+
+``` prolog
+:- meta_predicate(tfilter(2, ?, ?)).
+
+tfilter(_, [], []).
+tfilter(C_2, [E|Es], Fs0) :-
+   …
+```
+C_2 can be a partial goal requring two more arguments, or a 2-ary predicate.  In either case the parameter of C_2 is the reified
+truth value. 
+
+For example, using dif/3 from reif, and applying one argument 3, gives a partially applied goal X_2 requiring two more arguments.
+
+``` prolog
+?- X_2=dif(3), call(X_2,9,T).
+   X_2 = dif(3), T = true.
+?- 
+```
+or more succinctly:
+
+``` prolog
+?-  call(dif(3),9,T).
+   T = true.
+?- 
+```
+
+You will probably need to inspect reif.pl to fully understand how to use the module .  By studying the implementation of predicates, you may also learn
+some new prolog skills
+
+There is some code that you may find confusing, which shows a predicate expecting 2 arguments, but only one is provided:
+
+``` prolog
+   if_(call(C_2, E), Fs0 = [E|Fs], Fs0 = Fs),
+```
+Fortunatley, you don't need to understand that mechanism to use the reif module. The C_2 parameter name gives you enough information about the predicate
+you must supply  as an argument.
+
+### Surprises
+
+You may be surprised when something like this works:
+
+``` prolog
+?- call( tfilter,=(3),[3],X).
+   X = [3].
+?- 
+```
+
+but this breaks.
+
+``` prolog
+?- call( tfilter,#>(3),[3],X).
+   error(existence_error(procedure,(#>)/3),(#>)/3).
+?- 
+```
+
+What is happening is that reif has provided (=)/3, and that is what is being used in tfilter (not the usual (=)/2).   
+
+But (in my system at the time of writing) (#>)/2 exists, but there is no (#>)/3.  
+A workaround is to define a three-arity predicate to do the comparison (it could be (#>)/3, but I choose pound\_gt\_t):
+
+``` prolog
+pound_gt_t(X,Y,T) :- if_(Y #< X, T=true, T=false).
```

Comment:
In clpz, `(#<)/3` is defined as:

```prolog
#<(X, Y, T) :- clpz_t(X #< Y, T).
```

Probably better to define  `(#>)/3` analogously:

```prolog
#>(X, Y, T) :- clpz_t(X #> Y, T).
```